.gitlab/build_and_test.sh

```diff
@@ -0,0 +1,223 @@
+#!/bin/bash
+PROJECT_NAME=singularity-eos
+PROJECT_DEFAULT_BRANCH=main
+PROJECT_GROUP=oss
+BUILD_DIR=${BUILD_DIR:-build}
+SOURCE_DIR=${CI_PROJECT_DIR:-$PWD}
+PROJECT_SPACK_ENV=${PROJECT_SPACK_ENV:-/usr/projects/xcap/spack-env/oss/2024-08-01}
```

Comment:
I take it this will need updating whenever we have a rebuild that we want to leverage?